### PR TITLE
Send success messages only after DB-side operations

### PR DESCRIPTION
**Pull request by antepusic**
This PR fixes Bolt server operation by performing database operations like commit and rollback before sending the transaction success message. Linked to issue #1162.

[master < Task] PR
- [x] Check, and update documentation if necessary
- [x] Provide the full content or a guide for the final git message
> Send Bolt success messages only after DB operations run successfully

To keep docs changelog up to date, one more thing to do:
- [x] Write a release note here, including added/changed clauses
> We’ve fixed a bug about clients receiving a success message from Memgraph before certain DB-side operations were executed successfully.
- [x] Tag someone from docs team in the comments

### CODE REVIEW

File: src/communication/bolt/v1/states/handlers.hpp [hunk 1]
```diff
@@ -367,13 +367,13 @@ State HandleReset(TSession &session, const Marker marker) {
...
 
+  session.Abort();
+
   if (!session.encoder_.MessageSuccess()) {
     spdlog::trace("Couldn't send success message!");
     return State::Close;
   }
 
-  session.Abort();
-
```
[the remark this answers is not in the document]
Comment (andrejtonev) [reply src/communication/bolt/v1/states/handlers.hpp:376]: I think it is. 
Can you just put it in a try-catch, like we do with Commit?

**Review by andrejtonev — COMMENTED** [review]
Good work.
Two things are missing.
1) The try-catch Abort (left a comment)
2) `HandleBegin` still sends the success message before doing the actual work

**Review by andrejtonev — APPROVED** [review]
:muscle:

**Comment by andrejtonev** [comment]
@antepusic before you merge, please fix the tests. 
The new driver tests seem to be flaky. Not sure if you need to add a main or read results, consume.

**Comment by antepusic** [comment]
@vpavicic Release note in the original comment

**Pull request merged**